metron-analytics/metron-profiler/src/main/java/org/apache/metron/profiler/bolt/ProfileBuilderBolt.java

```diff
@@ -246,6 +247,8 @@ private void flush(Tuple tickTuple) {
     // clear the execution state to prepare for the next window
     executor.clearState();
 
+    profileConfig.getTickUpdate().forEach((var, expr) -> executor.assign(var, expr, new HashMap<>()));
```

Comment:
At this point in execution, during a flush operation, we don't have a telemetry message.  It is the tick tuple that drives the flush.  We won't have a telemetry message to offer to the 'tick' expressions.

With how you have it now, the 'tick' expressions will be able to reference all of the state being maintained by the 'init' and 'update' expressions.

The one missing piece that I can think of is that the 'tick' expressions cannot access the result of the 'result' expression.  It might be useful to offer that up to the 'tick' expressions.